src/python/pants/backend/project_info/list_roots.py

```diff
@@ -55,7 +55,7 @@ async def all_roots(source_root_config: SourceRootConfig) -> AllSourceRoots:
 @goal_rule
```

Comment:
@benjyw : Commenting here because I can't comment higher in the file, but: the `all_roots` rule isn't accounting for marker files: I think it will need to be updated to also match the marker file patterns in the initial snapshot, and should probably move out of this file nearby the other roots code.